doc/contributing.rst

```diff
@@ -248,6 +248,94 @@ or
 
 the first is preferred.
 
+Best practices
+=============
+
+Note: these are gneral guidelines, not hard rules; there are always exceptions.
+Code reviews can just point to a specific section here to save time and
+propagate best practices.
+
+.. _noimplicitbool:
+Take advantage of no implicit bool conversion
+
+.. code-block:: nim
+
+  doAssert isValid() == true
+  doAssert isValid() # preferred
+
+.. _noimplicitbool:
+Reduce parenthesis nesting
+
+.. code-block:: nim
+
+  doAssert (not 1 == 2)
+  doAssert: not 1 == 2  # preferred
+
+.. _immediately_invoked_lambdas:
+Immediately invoked lambdas (https://en.wikipedia.org/wiki/Immediately-invoked_function_expression)
+
+.. code-block:: nim
+
+  let a = (proc (): auto = getFoo())()
+  let a = block:  # preferred
+    getFoo()
+
+.. _design_for_ufcs:
+Design with UFCS (method call syntax) chaining in mind
+
+.. code-block:: nim
+
+  proc foo(cond: bool, lines: seq[string]) # bad
+  proc foo(lines: seq[string], cond: bool) # preferred
+  # can be called as: `getLines().foo(false)`
+
+.. _avoid_quit:
+Use exceptions (including assert / doAssert) instead of ``quit``
+rationale: https://forum.nim-lang.org/t/4089
+
+.. code-block:: nim
+
+  quit() # bad in almost all cases
+  doAssert() # preferred
+
+.. _tests_use_doAssert:
+Use ``doAssert`` (or ``require``, etc), not ``assert`` in all tests.
+
+.. code-block:: nim
+
+  runnableExamples: assert foo()
```

Comment:
Did you forget to add `# bad` and `# preferred` comments here?

Reply:
@kaushalmodi done; 
maybe you can use this: [github's new 'Suggested change' in code review is great; should be used - Nim forum](https://forum.nim-lang.org/t/4317)
I'm gonna start using it whenever possible in code reviews

Reply:
@timotheecour Yes, I saw that. It's still not clear how to use that.. I'll read up on that before using it. Thanks :)